### PR TITLE
ifp: Issue an error if -site is not specified and update documention.

### PR DESCRIPTION
**Pull request by titan73**

### CODE REVIEW

File: src/ifp/README.md [hunk 1]
```diff
@@ -52,30 +52,29 @@ die =  ( 0, 0 )
...
 initialize_floorplan
+  -die_area {llx lly urx ury} | -utilization util
   [-additional_sites site_names]
   [-aspect_ratio ratio]
   [-core_area {llx lly urx ury}]
   [-core_space space | {bottom top left right}]
-  [-die_area {llx lly urx ury}]
   [-flip_sites site_names]
   [-row_parity NONE|EVEN|ODD]
   [-site site_name]
```
Comment (maliberty) [inline src/ifp/README.md:62]: -site is mandatory

File: src/ifp/README.md [hunk 2]
```diff
@@ -52,30 +52,29 @@ die =  ( 0, 0 )
 
 ```tcl
 initialize_floorplan
+  -die_area {llx lly urx ury} | -utilization util
```
Comment (maliberty) [inline src/ifp/README.md:55]: The requirements are actually more complex.  You can either use (-utilization) or (-die_area and -core_area).

Comment (maliberty) [reply src/ifp/README.md:55]: I'm not sure why -utilization makes -core_space optional (0 by default) and -die_area requires -core_area.  It feels like both should either be required or not (probably required).  @rovinski @gadfort any opinions?

Comment (gadfort) [reply src/ifp/README.md:55]: I think die_area and core_area go together so if one is used the other should be required (it's possible to use core_space to derive the other if one wants to get fancy about it).
I would be okay saying core_space is required with utilization + aspect_ratio.

Comment (rovinski) [reply src/ifp/README.md:55]: (utilization && core_space) || (die_area && core_area)

Comment (maliberty) [reply src/ifp/README.md:55]: Look like a consensus.  Please make the change in this PR.

Comment (maliberty) [reply src/ifp/README.md]: Based on the above -core_space  should be mandatory not optional.  I don't see that reflected in your update.

**Comment by titan73** [comment]
PR updated.

File: src/ifp/README.md [hunk 3]
```diff
@@ -52,30 +52,26 @@ die =  ( 0, 0 )
 
 ```tcl
 initialize_floorplan
+  (-die_area {llx lly urx ury} -core_area {llx lly urx ury}) | (-utilization util [-aspect_ratio ratio] [-core_space (space | {bottom top left right})])
```
Comment (maliberty) [inline src/ifp/README.md:55]: `[-core_space` should drop the [ as it isn't optional.

Comment (titan73) [reply src/ifp/README.md:55]: -core_space is optional in current code. When no stecipied that core space is 0. Should I make it mandatory?
```
    if { [info exists keys(-core_space)] } {
      set core_sp $keys(-core_space)
      if { [llength $core_sp] == 1 } {
        sta::check_positive_float "-core_space" $core_sp
        set core_sp_bottom $core_sp
        set core_sp_top $core_sp
        set core_sp_left $core_sp
        set core_sp_right $core_sp
      } elseif { [llength $core_sp] == 4 } {
        lassign $core_sp core_sp_bottom core_sp_top core_sp_left core_sp_right
        sta::check_positive_float "-core_space" $core_sp_bottom
        sta::check_positive_float "-core_space" $core_sp_top
        sta::check_positive_float "-core_space" $core_sp_left
        sta::check_positive_float "-core_space" $core_sp_right
      } else {
        utl::error IFP 13 "-core_space is either a list of 4 margins or one value for all margins."
      }
    } else {
      set core_sp_bottom 0.0
      set core_sp_top 0.0
      set core_sp_left 0.0
      set core_sp_right 0.0
    }
```

Comment (maliberty) [reply src/ifp/README.md:55]: Yes I think it should be mandatory in the same with -core_area is.

Comment (titan73) [reply src/ifp/README.md]: Ok. Fixed.

File: src/ifp/README.md [hunk 4]
```diff
@@ -64,12 +64,12 @@ initialize_floorplan
 | Switch Name | Description |
 | ----- | ----- |
 | `-core_area` | Core area coordinates in microns (lower left x/y and upper right x/y coordinates). |
+| `-core_space` | Space around the core, default `0.0` microns. Allowed values are either one value for all margins or a set of four values, one for each margin. The order of the four values are: `{bottom top left right}`. |
```
Comment (maliberty) [inline src/ifp/README.md:67]: There is no `default 0.0 microns` anymore

**Review by maliberty — APPROVED** [review]

**Pull request merged**